Mathlib/RingTheory/Valuation/Integers.lean

```diff
@@ -133,6 +138,34 @@ theorem le_iff_dvd {x y : O} : v (algebraMap O F x) ≤ v (algebraMap O F y) ↔
   ⟨hv.dvd_of_le, hv.le_of_dvd⟩
 #align valuation.integers.le_iff_dvd Valuation.Integers.le_iff_dvd
 
+/--
+This is the special case of `Valuation.Integers.isUnit_of_one` when the valuation is defined
+over a field. Let `v` be a valuation on some field `F` and `O` be its integers. For every element
+`x` in `O`, `x` is a unit in `O` if and only if the image of `x` in `F` has valuation 1.
+-/
+theorem isUnit_of_one' {x : O} (hvx : v (algebraMap O F x) = 1) :
+    IsUnit x := by
+  apply isUnit_of_one hv _ hvx
+  apply IsUnit.mk0
+  rw [← v.ne_zero_iff]
+  simp only [hvx, ne_eq, one_ne_zero, not_false_eq_true]
+
+theorem eq_algebraMap_or_inv_eq_algebraMap (x : F) :
+    ∃ a : O, x = algebraMap O F a ∨ x⁻¹ = algebraMap O F a := by
```

Comment:
Can you use this to golf `Valuation.valuationSubring`?

Reply:
Sorry, what do you mean by "golf"?

Reply:
Shorten the proof of. As in [code golf](https://en.wikipedia.org/wiki/Code_golf).